the-collab-lab/tcl-79-smart-shopping-list
# Share list component

## Description

With the change it allows the users to invite others to existing shopping lists so they can manage them with friends or family. 

## Related Issue

Close #6 

## Acceptance Criteria

- [x]  The ManageList view shows a form that allows the user to enter an email to invite an existing user to a list, in addition to the form that allows them to add items to that list.
- [x]   If the other user does not exist, the user is shown an error message that explains the problem.


## Type of Changes

Feature

## Updates

### Before

![Screen Shot 2024-08-26 at 16 24 37](https://github.com/user-attachments/assets/80886ce4-fe56-418a-b6da-39404984618d)

### After

![Screen Shot 2024-08-26 at 16 27 49](https://github.com/user-attachments/assets/584b16cc-713d-4d8b-a548-b2d25bc6f588)

![Screen Shot 2024-08-26 at 16 30 57](https://github.com/user-attachments/assets/74807aed-f9c7-491f-8935-27bc958ca115)
![Screen Shot 2024-08-26 at 16 32 06](https://github.com/user-attachments/assets/d45ff27f-e42b-4eae-a750-cbe1237c1d6f)



## Testing Steps / QA Criteria

- Do a `git pull` and `git checkout  hm-mm-shareList`.
- Open the homepage by running npm start.
- Navigate to `ManageList` and you will find the `ShareListForm` where you can share your list with friends and family.
-  If the shopping list has been shared successfully a notification will appear telling the user this.
- Incase of an error sharing the shopping list an error notification will appear describing the error, telling the user to try again.
- To test the list sharing functionality, click on the specific list you would like to share on the `Home` page.
